tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/SDK/RuntimePipelineInvokeAsyncIntegration.cs

```diff
@@ -73,6 +73,12 @@ public class RuntimePipelineInvokeAsyncIntegration
                     // The request object is populated later by the Marshaller,
                     // so we wait until the method end callback to read it
                     tags.HttpMethod = executionContext.RequestContext.Request.HttpMethod.ToUpperInvariant();
+                    tags.HttpUrl = executionContext.RequestContext.Request.ResourcePath switch
+                    {
+                        null => executionContext.RequestContext.Request.Endpoint.AbsoluteUri,
+                        string resourcePath when resourcePath.Length > 0 && resourcePath[0] == '/' => executionContext.RequestContext.Request.Endpoint.AbsoluteUri + resourcePath.Substring(1),
+                        _ => executionContext.RequestContext.Request.Endpoint.AbsoluteUri + executionContext.RequestContext.Request.ResourcePath,
```

Comment:
Just to confirm, these values conform with the requirements for `http.url` in the latest RFC? (I can't remember what those requirements were, or know what these look like 😅)

Reply:
> For that reason, I feel like we should always be assigning a value to span.Tags["http.url"].

Sorry, what I really meant was that the value we're _assigning_ to `http.url` conforms with the upcoming spec. Because we currently use two different patterns for this tag (absolute URI, path only) +/- querystring! And as we're adding a new one, makes sense to do it "correctly", whichever way that is 🙂 

@anna-git, don't suppose you remember/have the RFC around the correct format?

Reply:
> don't suppose you remember/have the RFC around the correct format?

Spec is [here](https://datadoghq.atlassian.net/wiki/spaces/APM/pages/2357395856/Span+attributes#http.url).
Extract:  Full HTTP request URL in the form scheme://host[:port]/path[?query][#fragment]
Note: Query string should be captured by default. This is a change to previous behaviour requested by ASM, the reasons are presented in [ASM RFC - Capturing additional APM data](https://docs.google.com/document/d/1V-HkX3TW9ffhZ2-D7cmwWWEvp0IcUOnAnBCta_hTEIU/edit#heading=h.o5gstqo08gu5) which has been reviewed by the security advice group and legal.

Reply:
I've gotten confirmation from @robertpi that the RFC currently targets server spans. If the decision is to enforce the same standards on client spans, we can address that in a follow-up PR.